clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3518,6 +3545,89 @@ private void produceToLocalKafkaHelper(
+      String brokerUrl,
+      long beforeProcessingRecordTimestampNs,
+      LeaderMetadataWrapper leaderMetadataWrapper,
+      LeaderProducedRecordContext leaderProducedRecordContext) {
+    final InternalAvroSpecificSerializer<GlobalRtDivState> serializer =
+        AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getSerializer(); // TODO: can't be static because of state?
+    final byte[] keyBytes = getGlobalRtDivKeyBytes(brokerUrl);
+    TopicType realTimeTopicType = TopicType.of(TopicType.REALTIME_TOPIC_TYPE, brokerUrl);
+
+    // Snapshot the VT DIV + RT DIV (single broker URL) in preparation to be produced
+    PartitionTracker divClone = kafkaDataIntegrityValidatorForLeaders.cloneProducerStates(partition, brokerUrl);
+    Map<CharSequence, ProducerPartitionState> rtDiv = divClone.getPartitionStates(realTimeTopicType);
+
+    // Create GlobalRtDivState (RT DIV + latestOffset) which will be serialized into a byte array
+    ByteBuffer emptyBuffer = ByteBuffer.allocate(0);
+    GlobalRtDivState globalRtDiv = new GlobalRtDivState(brokerUrl, rtDiv, previousMessage.getOffset(), emptyBuffer);
+
+    // Create PubSubMessage for the LeaderProducerCallback to enqueue the RT + VT DIV to the drainer
+    KafkaKey divKey = new KafkaKey(MessageType.GLOBAL_RT_DIV, keyBytes);
+    // TODO: incrementSequenceNumber? are the pubsubmessage fields correct?
+    KafkaMessageEnvelope divEnvelope = getVeniceWriter(partitionConsumptionState).get()
+        .getKafkaMessageEnvelope(
+            MessageType.PUT,
+            false,
+            partition,
+            true,
+            DEFAULT_LEADER_METADATA_WRAPPER,
+            APP_DEFAULT_LOGICAL_TS);
+    divEnvelope.payloadUnion = new DivSnapshot(divClone, previousMessage.getOffset()); // contains both VT + RT DIV
+    PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> divMessage = new ImmutablePubSubMessage<>(
+        divKey,
+        divEnvelope,
+        previousMessage.getTopicPartition(),
+        previousMessage.getOffset(), // TODO: are these reused fields correct?
+        previousMessage.getPubSubMessageTime(),
+        divKey.getHeapSize()); // TODO: should the envelope size also be estimated?
+    LeaderProducerCallback divCallback = createProducerCallback(
+        divMessage,
+        partitionConsumptionState,
+        leaderProducedRecordContext,
+        partition,
+        brokerUrl,
+        beforeProcessingRecordTimestampNs);
+
+    // Produce to local kafka for the Global RT DIV + latestOffset (GlobalRtDivState)
+    getVeniceWriter(partitionConsumptionState).get()
+        .put(
+            keyBytes,
+            ByteUtils.extractByteArray(serializer.serialize(globalRtDiv)),
+            partition,
+            1, // dummy value schema id which shouldn't be used for MessageType.GLOBAL_RT_DIV
+            divCallback,
+            leaderMetadataWrapper,
+            APP_DEFAULT_LOGICAL_TS,
+            null,
+            null, // TODO: do oldValueManifest and rmd need to be populated?
```

Comment:
Since this will likely going to be chunked, then we should as well get the old manifest and perform the actual deletion.